### ISSUE TITLE
Deprecated Functionality: Function mcrypt_module_open() is deprecated with php:7.1-fpm-0

### ISSUE THREAD

**Issue by iandd0824** [mageinferno/magento2-docker-compose#54]
Try to install the version 2.1.4 and run into the deprecated exception if I use the php version magento2-php:7.1-fpm-0 in setup container.

There is no deprecated exception with magento2-php:7.0-fpm-0.

docker-compose.yml
```
setup:
  image: mageinferno/magento2-php:7.1-fpm-0
  command: /usr/local/bin/mage-setup
  links:
    - db
  volumes_from:
    - appdata
  env_file: env/setup.env
```

Command
```
$ docker-compose run --rm setup
```

Deprecated Exception
```
[Progress: 114 / 425]
Module 'Magento_Customer':
[Progress: 115 / 425]
Module 'Magento_BundleImportExport':
[Progress: 116 / 425]
Module 'Magento_CacheInvalidate':
[Progress: 117 / 425]
Module 'Magento_AdminNotification':
[Progress: 118 / 425]
Module 'Magento_Indexer':
Running schema recurring...


[Exception] Deprecated Functionality: Function mcrypt_module_open() is deprecated in /var/www/html/vendor/magento/framework/Encryption/Crypt.php on line 54

```

**Comment by jackbrandworkz** [comment]
+1
Solved issue temporarily by using PHP FHM 7.0 image in the docker-compose.yml file

PLEASE APPLY THE FOLLOUWNG FIX - 
https://www.cadence-labs.com/2016/09/magento-2-php-7-1-fix-function-mcyrpt_module_open-deprecated/

SOLUTION (https://www.cadence-labs.com/wp-content/themes/Divi-reddot/gists/Crypt.php.txt)

**Comment by markoshust** [comment]
Not going to monkey-patch Magento 2's shit core code. They can do that.

I merged in a fix that downgrades back to PHP 7.0. Once @magento cares enough to put out a proper fix (and not take months to do it), I'll upgrade the compose back to PHP 7.1.
